SU-SWS/stanford_profile
# Configure the new embeddable validator

# READY FOR REVIEW

# Summary
- Configure the embeddable validator from [stanford_media](https://github.com/SU-SWS/stanford_media/pull/103)
- This allows site managers the ability to create their embedded content if we have a validation plugin to match. We'll have to add more plugins as we vet the providers, but admins still have full access.

# Need Review By (Date)
- 9/24

# Urgency
- High

# Steps to Test
1. checkout this branch and the matching branch in [stanford_media](https://github.com/SU-SWS/stanford_media/pull/103)
2. `blt drupal:update`
3. use masquerade or disable saml and log in as a contributor
4. navigate to `/media/add/embeddable` and verify you still don't ge tthe full embed code field
5. log out and log in as a site manager
6. navigate to `/media/add/embeddable` and try to put some random code or text in the embed code field
7. when you save, verify you get an error
8. Enter a localist widget embed code like:
```
<div id="localist-widget-88041469" class="localist-widget"></div>
<script defer type="text/javascript" src="http://stanford.enterprise.localist.com/widget/view?schools=stanford&days=31&num=50&container=localist-widget-88041469&template=modern"></script>
<a href="http://stanford.edu">Stanford</a>
```
9. Save and verify it does indeed save
10. edit the created media item
11. verify that the `<a>` tag from the above code was stripped from the embed code.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
